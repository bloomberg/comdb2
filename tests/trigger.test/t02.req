CREATE TABLE t2 (x INTEGER)$$
CREATE TABLE t2_1 (x INTEGER)$$
INSERT INTO t2_1 VALUES (1)

CREATE PROCEDURE trigger_2 VERSION 'trigger_2' {
  local function main(event)
    db:prepare("select * from t2_1"):exec()
    local tbl = db:table("tbl", {{"i", "int"}})
    local tp = event.type
    local inew, iold
    if tp == 'add' then
      inew = event.new.x
      tbl:insert({i=inew})
    elseif tp == 'del' then
      iold = event.old.x
      tbl:insert({i=iold})
    end
    return 0
  end
}$$

CREATE LUA TRIGGER trigger_2 ON (TABLE t2 FOR INSERT AND UPDATE AND DELETE)
SELECT SLEEP(1) -- let trigger start
INSERT INTO t2 VALUES(1)
SELECT SLEEP(1) -- let trigger fire
SELECT 'the alter-table statement should not hang'
ALTER TABLE t2_1 ADD COLUMN y INTEGER$$
SELECT * FROM t2_1 -- make sure schema change succeeded
DROP LUA TRIGGER trigger_2
