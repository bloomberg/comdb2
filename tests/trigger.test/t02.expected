(rows inserted=1)
(version='trigger_2')
(SLEEP(1) -- let trigger start=1)
(rows inserted=1)
(SLEEP(1) -- let trigger fire=1)
('the alter-table statement should not hang'='the alter-table statement should not hang')
(x=1, y=NULL)
